parachain/runtime/kintsugi/src/dex.rs

```diff
@@ -0,0 +1,243 @@
+    KINT, KSM,
+};
+use sp_runtime::{traits::Zero, FixedPointNumber};
+
+#[cfg(feature = "try-runtime")]
+use frame_support::ensure;
+
+pub use dex_general::{AssetBalance, GenerateLpAssetId, PairInfo};
+pub use dex_stable::traits::{StablePoolLpCurrencyIdGenerate, ValidateCurrency};
+
+parameter_types! {
+    pub const DexGeneralPalletId: PalletId = PalletId(*b"dex/genr");
+    pub const DexStablePalletId: PalletId = PalletId(*b"dex/stbl");
+    pub const StringLimit: u32 = 50;
+}
+
+pub struct PairLpIdentity;
+impl GenerateLpAssetId<CurrencyId> for PairLpIdentity {
+    fn generate_lp_asset_id(asset_0: CurrencyId, asset_1: CurrencyId) -> Option<CurrencyId> {
+        CurrencyId::join_lp_token(asset_0, asset_1)
+    }
+}
+
+impl dex_general::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type MultiCurrency = Tokens;
+    type PalletId = DexGeneralPalletId;
+    type AssetId = CurrencyId;
+    type LpGenerate = PairLpIdentity;
+    type WeightInfo = ();
+}
+
+pub struct PoolLpGenerate;
+impl StablePoolLpCurrencyIdGenerate<CurrencyId, StablePoolId> for PoolLpGenerate {
+    fn generate_by_pool_id(pool_id: StablePoolId) -> CurrencyId {
+        CurrencyId::StableLpToken(pool_id)
+    }
+}
+
+pub struct StableAmmVerifyPoolAsset;
+impl ValidateCurrency<CurrencyId> for StableAmmVerifyPoolAsset {
+    fn validate_pooled_currency(_currencies: &[CurrencyId]) -> bool {
+        true
+    }
+
+    fn validate_pool_lp_currency(currency_id: CurrencyId) -> bool {
+        if Tokens::total_issuance(currency_id) > 0 {
+            return false;
+        }
+        true
+    }
+}
+
+impl dex_stable::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type CurrencyId = CurrencyId;
+    type MultiCurrency = Tokens;
+    type PoolId = StablePoolId;
+    type TimeProvider = Timestamp;
+    type EnsurePoolAsset = StableAmmVerifyPoolAsset;
+    type LpGenerate = PoolLpGenerate;
+    type PoolCurrencySymbolLimit = StringLimit;
+    type PalletId = DexStablePalletId;
+    type WeightInfo = ();
+}
+
+impl dex_swap_router::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type StablePoolId = StablePoolId;
+    type Balance = Balance;
+    type CurrencyId = CurrencyId;
+    type NormalAmm = DexGeneral;
+    type StableAMM = DexStable;
+    type WeightInfo = ();
+}
+
+pub struct DexSetup;
+
+fn create_pair_and_update_rewards(asset_0: CurrencyId, asset_1: CurrencyId, rewards: Balance, fee_rate: Balance) {
+    if let Err(err) = DexGeneral::create_pair(RuntimeOrigin::root(), asset_0, asset_1, fee_rate) {
+        log::error!("Could not create pair: {:?}", err);
+    } else {
+        if let Err(err) = Farming::update_reward_schedule(
+            RuntimeOrigin::root(),
+            CurrencyId::join_lp_token(asset_0, asset_1).expect("currencies are valid; qed"),
+            CurrencyId::Token(KINT),
+            132480, // 60 * 24 * 92 (approx three months)
```

Comment:
It actually says so in the discussion post (3 month -> https://kintsugi.subsquare.io/post/15) and I also stated this in the draft for the governance proposal so that should be fine.